src/main/java/seedu/address/ui/StatisticsPanel.java

```diff
@@ -13,79 +12,263 @@
+        int monthCount = 0;
+
+        ArrayList<Integer> monthPersonsAdded = getNewPersonsAddByMonth(list);
+
+        for (int i = startYear; i <= endYear; i++) {
+
+            startMonth = 1;
+            endMonth = 12;
+
+            if (i == startYear) {
+                startMonth = this.currentMonth;
+            }
+
+            if (i == endYear) {
+                endMonth = this.currentMonth;
+            }
+
+            for (int m = startMonth; m <= endMonth; m++) {
+
+                String labelName = Month.of(m).name().substring(0, 3) + " " + Integer.toString(i);
+                aSeries.getData().add(new XYChart.Data(labelName, monthPersonsAdded.get(monthCount)));
+
+                monthCount++;
+            }
+        }
+        answer.addAll(aSeries);
+        return answer;
     }
 
+    private ArrayList<Integer> getNewPersonsAddByMonth(ObservableList<ReadOnlyPerson> list) {
+
+
+        ArrayList<Integer> countByMonth = new ArrayList<>(Collections.nCopies(PERSON_ADDED_DISPLAY_YEARS * 12 + 1, 0));
+
+        list.forEach((p) -> {
+            Date givenDate = p.getCreatedAt();
+            ZonedDateTime given = givenDate.toInstant().atZone(ZoneId.of("UTC"));
+
+            int personAddedYear = Integer.parseInt(Year.from(given).toString());
+            int personAddedMonth = Month.from(given).getValue();
+
+            int indOffset = calculateCountByMonthOffset(personAddedMonth, personAddedYear);
+            if ( indOffset >= 0 && indOffset <= PERSON_ADDED_DISPLAY_YEARS * 12) {
+                countByMonth.set(indOffset, countByMonth.get(indOffset) + 1);
+            }
+        });
+
+        return countByMonth;
+    }
     /**
-     * Formats the data into PieChart.Data for display
+     * Count the offset when adding to the array list of sum by months
      */
-    private ObservableList<PieChart.Data> tabulateAddedByYear(ObservableList<ReadOnlyPerson> list) {
+    public int calculateCountByMonthOffset(int personAddedMonth, int personAddedYear) {
+        return (this.currentYear - personAddedYear) * 12 +
+                (this.currentMonth - personAddedMonth);
+    }
 
-        ArrayList<PieChart.Data> data = new ArrayList<>();
+    /**
+     * Tabulate the total number of people in the list
+     */
+    public void tabulateTotalNumberOfPeople(ObservableList<ReadOnlyPerson> list) {
+        this.totalNumberOfPeople = list.size();
+    }
 
-        ArrayList<Integer> yearData = collectYear(list);
-        for (int y = 0; y < yearData.size(); y++) {
-            if (yearData.get(y) > 0) {
-                String yearLabel = Integer.toString(y + PERSON_ADDED_EARLIEST_YEAR)
-                        + " - " + yearData.get(y);
-                data.add(new PieChart.Data(yearLabel, yearData.get(y)));
+    /**
+     * Tabulates number of users of each social media platform
+     */
+    public void tabulateSocialMediaUsage(ObservableList<ReadOnlyPerson> list) {
+        for (ReadOnlyPerson aList : list) {
```

Comment:
Will it be better to use the variable name "person" instead of "aList"? I'm just nitpicking haha

Reply:
yeah sure thanks